src/metabase/sync/analyze/classifiers/name.clj

```diff
@@ -87,3 +120,36 @@
                        (sync-util/name-for-logging field)
                        inferred-special-type))
     (assoc field :special_type inferred-special-type)))
+
+(def ^:private entity-types-patterns
+  [[#"order"        :entity/TransactionTable]
+   [#"transaction"  :entity/TransactionTable]
+   [#"sale"         :entity/TransactionTable]
+   [#"product"      :entity/ProductTable]
+   [#"user"         :entity/UserTable]
+   [#"account"      :entity/UserTable]
+   [#"people"       :entity/UserTable]
+   [#"person"       :entity/UserTable]
+   [#"event"        :entity/EventTable]
+   [#"checkin"      :entity/EventTable]
+   [#"log"          :entity/EventTable]
+   [#"subscription" :entity/SubscriptionTable]
+   [#"company"      :entity/CompanyTable]
+   [#"companies"    :entity/CompanyTable]
+   [#"vendor"       :entity/CompanyTable]])
+
+(s/defn ^:always-validate infer-entity-type :- i/TableInstance
+  [table :- i/TableInstance]
```

Comment:
`s/defn` generates basic docstrings which is why the linter didn't complain but it would still be nice to have a brief blurb explaining what exactly is going on here